portfolio/src/main/java/com/google/sps/servlets/DeletionServlet.java

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.io.IOUtils;
+
+/** Servlet that returns a programmable number of comments */
+@WebServlet("/delete-comment")
+public class DeletionServlet extends HttpServlet {
+
+  /**
+    * POST handler for the /delete-comment route
+    */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    // get the body of the request
+    String bodyString = IOUtils.toString(request.getReader());
+
+    // parse the json request body and create a DeletionRequestBody from its contents
+    Gson gson = new Gson();
+    DeletionRequestBody body =
+        gson.fromJson(bodyString, DeletionRequestBody.class);
```

Comment:
What should bodyString look like? Is there a datastructure for it?

Reply:
It should be a json encoded DeletionRequestBody. I'm going to improve commenting now about this. I can see where it falls short in terms of adequately describing what bodyString is expected to look like.

Reply:
DeletionRequestBody is a private inner class inside of DeletionServlet (on lines 71+)

Reply:
I've added some better documentation for this https://github.com/thomasogara/step-repository/pull/34/commits/1f91aa9f0662c1d5201e4c74c8f7497f0d86867b do let me know if there's anything about it that doesn't read well

Reply:
Yes that looks fine, feel free to commit that change into this branch